### PR TITLE
Create notifications page

### PR DESCRIPTION
**Pull request by avalucianelson**
## Description

I created a "Notifications" page based on the High Fidelity Figma Designs, which can be found [here](https://www.figma.com/design/vnyb5qwvnjpDVaO7rOC3ZL/ID8-High-Fidelity-Designs?node-id=0-1&node-type=canvas&t=njuFGgzYL6DFwx1B-0). I'm currently using hardcoded data, but it should be easily replaced with the data from the backend. Please let me know if that is not the case or if there's any way I can help with that process :) 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tests (adding or updating tests)
- [ ] Documentation update

## Testing Instructions

To test it out, run the code and navigate to /notifications 

## Screenshots (if applicable)

Here's what the Figma design's notifications screen looks like: 
image

Here's what my code's notifications screen looks like: 
image

I know there are some slight differences. For now, I wanted to focus on getting it close enough since we're so close to launch, and then can reiterate on designs later. The only creative liberty I took (that wasn't because I couldn't figure out how to do something) was changing the text color from a light gray to a slightly darker gray. I felt it made it easier to read and potentially more accessible. I also adjusted uses of "startup idea" to "project" in line with the changes we've been making for Minervans.

### CODE REVIEW

**Review by FlaIespa — COMMENTED** [review]
Great work, Ava! Check out my comments below :)

File: src/app/notifications/page.tsx
Comment (FlaIespa) [inline src/app/notifications/page.tsx:1]: I understand that you have worked on the Navbar as part of another PR, correct? When starting this new branch for the notifications page, did you rebase on the Navbar branch? Here is the reason why I ask: 
I don't believe it is good practice to rebase on feature branches, as that creates a bit of unstability in the repo. I believe that we want to always branch out from main, ensure that main is on its latest version, and work from there. Once you return to work, you want to fetch from the main remote and rebase on main to ensure that if any PR has been merged, you get those updates for when you push your local and try to merge there are no merge conflicts. 
The reason why I am saying all of this is because I can't see the Navbar component when checking out the ava/notifications-page branch. I believe that is because that lives in another PR and when looking at Git Graph I don't see a common history between your branches ava/notifications-page and ava/navbar-update. 
Because there is no Navbar component in this PR, when I try to run the app from this branch and move to /notifications as per the PR instructions, I get a `Module not found: Can't resolve '../_components/Navbar'` error. 
I might be missing something here given that I see the screenshot you added in the PR with the functioning page, so please feel free to clarify and call out if I am missing something!

File: src/app/notifications/page.tsx
Comment (FlaIespa) [inline src/app/notifications/page.tsx:5]: I understand that you have worked on the Navbar as part of another PR, correct? When starting this new branch for the notifications page, did you rebase on the Navbar branch? Here is the reason why I ask: 
I don't believe it is good practice to rebase on feature branches, as that creates a bit of unstability in the repo. I believe that we want to always branch out from main, ensure that main is on its latest version, and work from there. Once you return to work, you want to fetch from the main remote and rebase on main to ensure that if any PR has been merged, you get those updates for when you push your local and try to merge there are no merge conflicts. 
The reason why I am saying all of this is because I can't see the Navbar component when checking out the ava/notifications-page branch. I believe that is because that lives in another PR and when looking at Git Graph I don't see a common history between your branches ava/notifications-page and ava/navbar-update. 
Because there is no Navbar component in this PR, when I try to run the app from this branch and move to /notifications as per the PR instructions, I get a `Module not found: Can't resolve '../_components/Navbar'` error. 
I might be missing something here given that I see the screenshot you added in the PR with the functioning page, so please feel free to clarify and call out if I am missing something! 
Given that the navbar is in the other PR, we can merge that one, rebase this branch on main and then we are going to get the navbar component. Does this make sense to you?

File: src/app/notifications/page.tsx
Comment (FlaIespa) [inline src/app/notifications/page.tsx:13]: Using a mock array is a nice approach for development and testing, as we want to see the notifications when running the app and we have no connection with the backend. However, given that the goal is to connect with the backend asap, I would suggest commenting it out or removing it, as we are not using this in production.

File: src/app/notifications/page.tsx
Comment (FlaIespa) [inline src/app/notifications/page.tsx:42]: I could see how the notifications could be broken down into individual components, but we can ignore this for now. It looks good this way too!

File: src/app/notifications/page.tsx
Comment (matviikotolyk) [reply src/app/notifications/page.tsx:5]: @FlaIespa I'm getting the same error. The easiest solution is probably to merge the PR where the NavBar is introduced, stash the changes on this branch, pull the latest from the main branch, and then pop the stash with the changes on this branch. Let me know if this makes sense!

**Review by FlaIespa — CHANGES_REQUESTED** [review]
Great work with the addition of the notifications page! I left a few comments we should address before merging. 
For the future, when completing the section of "Testing Instructions", make sure to include the name of the branch we are meant to check out before navigating to the change you proposed. Does that make sense? 
Thank you for all the hard work you have been putting in to make ID8 come to life!

File: src/app/notifications/page.tsx
Comment (FlaIespa) [reply src/app/notifications/page.tsx:5]: I believe that she won't need to stash changes, given that all the work has been committed and pushed. I would say then maybe just merge the Navbar into the main, pull the latest from the main in her local, and then rebase this branch on the main. What do you think @matviikotolyk? But I would also be up to trying your approach!

Comment (matviikotolyk) [reply src/app/notifications/page.tsx:5]: Oh yeah, that works. In my head, the changes were still on Ava's local machine... It seems like the PR with the Navbar looks good, so we can probably go ahead and merge it into main. @FlaIespa

Comment (FlaIespa) [reply src/app/notifications/page.tsx:5]: I just reviewed and it looks good to me too. I would just wait for Ava to merge it. Even though there does not seem to be any outstanding changes that need to be made,  there is still one unresolved conversation, so it might be better to let her address it. What do you think?

Comment (matviikotolyk) [reply src/app/notifications/page.tsx:5]: Totally, sounds good to me!

Comment (FlaIespa) [reply src/app/notifications/page.tsx:5]: I will resolve this so we can merge it! I rebased it on main and then force-pushed with lease.

**Review by FlaIespa — APPROVED** [review]

**Pull request merged**
